src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java

```diff
@@ -1,14 +1,177 @@
 package uk.gov.hmcts.reform.roleassignment.domain.service.deleteroles;
 
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.roleassignment.data.roleassignment.RequestEntity;
+import uk.gov.hmcts.reform.roleassignment.domain.model.AssignmentRequest;
+import uk.gov.hmcts.reform.roleassignment.domain.model.Request;
+import uk.gov.hmcts.reform.roleassignment.domain.model.RequestedRole;
+import uk.gov.hmcts.reform.roleassignment.domain.model.RoleAssignment;
+import uk.gov.hmcts.reform.roleassignment.domain.model.enums.Status;
+import uk.gov.hmcts.reform.roleassignment.domain.service.common.ParseRequestService;
+import uk.gov.hmcts.reform.roleassignment.domain.service.common.PersistenceService;
+import uk.gov.hmcts.reform.roleassignment.domain.service.common.ValidationModelService;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Service
 public class DeleteRoleAssignmentOrchestrator {
-    //1. call parse request service
-    //2. Call persistence service to store the created records
-    //3. Call Validation model service to create aggregation objects and apply drools validation rule
-    //5. For Each: If success then call persistence service to update assignment record status
-    //6. once all the assignment records are approved call persistence to update request status
-    //7. Call persistence to move assignment records to Deleted status
-    //8. Call the persistence to remove assignment records from RoleAssignmentLive table
+
+
+    private PersistenceService persistenceService;
+    private ParseRequestService parseRequestService;
+    private ValidationModelService validationModelService;
+    RequestEntity requestEntity;
+    AssignmentRequest assignmentRequest;
+
+    public DeleteRoleAssignmentOrchestrator(PersistenceService persistenceService,
+                                            ParseRequestService parseRequestService,
+                                            ValidationModelService validationModelService) {
+        this.persistenceService = persistenceService;
+        this.parseRequestService = parseRequestService;
+        this.validationModelService = validationModelService;
+    }
+
+    public ResponseEntity<Object> deleteRoleAssignment(String actorId,
+                                                       String process,
+                                                       String reference) throws Exception {
+        List<RequestedRole> requestedRoles = null;
+
+        //1. create the request Object
+        Request request = parseRequestService.prepareRequestObject(process, reference);
```

Comment:
what if process or reference are empty? we need to use this method only when both are available and throw 400 bad request if any one of them is not provided.

Reply:
Done.